### PR TITLE
Add basic Swift support (requires Xcode 6)

### PR DESCRIPTION
**Pull request by mblsha**
By default expressions are evaluated in current frame's language.

When stopped inside a Swift function, current language is Swift, and it's invalid to use Objective-C syntax: many Chisel functions break as a result.

Explicitly specify Objective-C++ as expression language when evaluating commands.

However one problem will still persist: it's still impossible to use Swift objects by name in Chisel functions. One needs to first convert object names to id pointers using evaluateExpressionValueInFrameLanguage(), but the lldb code to enable that was committed very recently and is still unavailable in Xcode 6.1.1.

### CODE REVIEW

**Comment by kastiglione** [comment]
Thank you @mblsha :octocat: 

Would you mind signing Facebook's Contributor License Agreement? If you have any questions I'd be happy to help answer. https://code.facebook.com/cla

**Comment by mblsha** [comment]
Ok, done ;-)

File: fblldbbase.py [hunk 1]
```diff
@@ -36,14 +36,27 @@ def run(self, arguments, option):
     pass
 
 
-def evaluateExpressionValue(expression, printErrors=True):
+def evaluateExpressionValueWithLanguage(expression, language, printErrors):
   # lldb.frame is supposed to contain the right frame, but it doesnt :/ so do the dance
   frame = lldb.debugger.GetSelectedTarget().GetProcess().GetSelectedThread().GetSelectedFrame()
-  value = frame.EvaluateExpression(expression)
+  expr_options = lldb.SBExpressionOptions()
+  expr_options.SetLanguage(language)  # requires lldb r210874 (2014-06-13) / Xcode 6
```
Comment (kastiglione) [inline fblldbbase.py]: Hey @arigrant, what do you think about baseline support? Should Chisel support anything more than just the current version of Xcode?

File: fblldbbase.py [hunk 2]
```diff
@@ -36,14 +36,27 @@ def run(self, arguments, option):
...
+  value = frame.EvaluateExpression(expression, expr_options)
   if printErrors and value.GetError() is not None and str(value.GetError()) != 'success':
     print value.GetError()
   return value
 
+def evaluateExpressionValueInFrameLanguage(expression, printErrors=True):
+  # lldb.frame is supposed to contain the right frame, but it doesnt :/ so do the dance
+  frame = lldb.debugger.GetSelectedTarget().GetProcess().GetSelectedThread().GetSelectedFrame()
+  language = frame.GetCompileUnit().GetLanguage()  # requires lldb r222189 (2014-11-17)
```
Comment (kastiglione) [inline fblldbbase.py]: @mblsha Which version of Xcode includes `r222189`?

**Comment by kastiglione** [comment]
@mblsha I've left a couple comments. Thanks for submitting this pull request!

The only thing that needs to be figured out Xcode/lldb version support.

File: fblldbbase.py (hunk 2 above)
Comment (mblsha) [inline fblldbbase.py]: Not sure, it's not yet in 6.1.1, so `evaluateExpressionValueInFrameLanguage()` is not actually called anywhere and is present for future reference. Hopefully LLDB in 6.2 beta has it, but I haven't verified it.

I'll try it once it becomes final.

**Comment by mblsha** [comment]
`evaluateExpressionValueWithLanguage()` works fine in Xcode 6.0+. Once 6.2 becomes final maybe `evaluateExpressionValueInFrameLanguage()` will work as well.

File: fblldbbase.py (hunk 2 above)
Comment (kastiglione) [inline fblldbbase.py]: > `evaluateExpressionValueInFrameLanguage()` is not actually called anywhere

:+1:

**Comment by moreindirection** [comment]
It looks like there are still a few problems in Swift. Any commands that are of the form "po EXPRESSION" need to be changed to "expression -O -l objc++ --- EXPRESSION"

Would really like to see these changes merged into master, though.

**Comment by kastiglione** [comment]
:ship: :ship: :ship: Thanks @mblsha!

**Comment by kastiglione** [comment]
@moreindirection Can I interest you in opening a pull request to fix the stuff you've seen broken with Swift? :cake:

**Comment by moreindirection** [comment]
@kastiglione Sure, I just created pull request #73

**Pull request merged**